File: core/product.py
import json
import boto3
import decimal

dynamodb = boto3.resource('dynamodb')
table = dynamodb.Table('dev-demostore-product')

# LoadData Function
def load(event, context):
    body = {
        "message": "ok load"
    }

    with open("core/moviedata.json") as json_file:
        movies = json.load(json_file, parse_float = decimal.Decimal)
        for movie in movies:
            year = int(movie['year'])
            title = movie['title']
            info = movie['info']
        
            table.put_item(
               Item={
                   'year': year,
                   'productID': title,
                   'info': info,
                }
            )

    response = {
        "statusCode": 200,
        "body": json.dumps(body)
    };

    return response


def hello(event, context):
    body = {
        "message": "Go Serverless v1.0! Your function executed successfully!",
        "input": event
    }

    response = {
        "statusCode": 200,
        "body": json.dumps(body)
    };

    return response

File: core/product.js
'use strict';

const aws = require('aws-sdk');

module.exports.hello = (event, context, callback) => {
  const response = {
    statusCode: 200,
    body: JSON.stringify({
      message: 'Go Serverless v1.0! Your function executed successfully!',
      input: event,
    }),
  };

  callback(null, response);
};


module.exports.view = (event, context, callback) => {
        
    var docClient = new aws.DynamoDB.DocumentClient()
    
    var table = "dev-demostore-product";
    var title = event.path.urlkey;
    
    var params = {
        TableName: table,
        Key:{
            "productID": title
        }
    };

    docClient.get(params, function(err, data) {
        if (err) {
            console.error("Unable to read item. Error JSON:", JSON.stringify(err, null, 2));
        } else {
            console.log("GetItem succeeded:", JSON.stringify(data, null, 2));
            callback(null, JSON.stringify(data, null, 2));
        }
    }); 

};
